bindings.go

```diff
@@ -0,0 +1,193 @@
+// OVS uuid -> go strings
+// OVS map  -> go map
+// OVS enum -> go native type depending on the type of the enum key
+func NativeType(column *ColumnSchema) (reflect.Type, error) {
+	switch column.Type {
+	case TypeInteger, TypeReal, TypeBoolean, TypeUUID, TypeString:
+		return nativeTypeFromExtended(column.Type)
+	case TypeEnum:
+		return nativeTypeFromExtended(column.TypeObj.Key.Type)
+	case TypeMap:
+		kType, err := nativeTypeFromExtended(column.TypeObj.Key.Type)
+		if err != nil {
+			return reflect.TypeOf(nil), err
+		}
+		vType, err := nativeTypeFromExtended(column.TypeObj.Value.Type)
+		if err != nil {
+			return reflect.TypeOf(nil), err
+		}
+		return reflect.MapOf(kType, vType), nil
+	case TypeSet:
+		kType, err := nativeTypeFromExtended(column.TypeObj.Key.Type)
+		if err != nil {
+			return reflect.TypeOf(nil), err
+		}
+		return reflect.SliceOf(kType), nil
+	default:
+		return reflect.TypeOf(nil), fmt.Errorf("Failed to determine type of column %s", column)
+	}
+}
+
+// OvsToNative transforms an ovs type to native one based on the column type information
+func OvsToNative(column *ColumnSchema, ovsElem interface{}) (interface{}, error) {
+	naType, err := NativeType(column)
+	if err != nil {
+		return nil, err
+	}
+	switch column.Type {
+	case TypeInteger, TypeReal, TypeString, TypeBoolean, TypeEnum:
+		// Atomic types should have the same underlying type
+		return ovsElem, nil
+	case TypeUUID:
+		uuid, ok := ovsElem.(UUID)
+		if !ok {
+			return nil, fmt.Errorf("Element in column should be convertible to UUID. Instead got %s", reflect.TypeOf(ovsElem))
+		}
+		return uuid.GoUUID, nil
+	case TypeSet:
+		// The inner slice is []interface{}
+		// We need to convert it to the real type os slice
+		nativeSet := reflect.MakeSlice(naType, 0, 0)
+
+		// RFC says that for a set of exactly one, an atom type an be sent
+		switch ovsElem.(type) {
+		case OvsSet:
+			ovsSet := ovsElem.(OvsSet)
+			for _, v := range ovsSet.GoSet {
+				vv, err := NativeValueOf(v, column.TypeObj.Key.Type)
+				if err != nil {
+					return nil, err
+				}
+				nativeSet = reflect.Append(nativeSet, vv)
+			}
+
+		default:
+			keyType, err := nativeTypeFromExtended(column.TypeObj.Key.Type)
+			if err != nil {
+				return nil, err
+			}
+
+			vv, err := NativeValueOf(ovsElem, column.TypeObj.Key.Type)
+			if err != nil {
+				return nil, err
+			}
+
+			if !vv.Type().ConvertibleTo(keyType) {
+				return nil, fmt.Errorf("Element in column should be convertible to Set of %s. Instead got %s", column.TypeObj.Key.Type, reflect.TypeOf(ovsElem))
+			}
+			nativeSet = reflect.Append(nativeSet, vv)
+		}
+		return nativeSet.Interface(), nil
+
+	case TypeMap:
+		ovsMap, ok := ovsElem.(OvsMap)
+		if !ok {
+			return nil, fmt.Errorf("Element in column should be convertible to Map. Instead got %s", reflect.TypeOf(ovsElem))
+		}
+		// The inner slice is map[interface]interface{}
+		// We need to convert it to the real type os slice
+		nativeMap := reflect.MakeMap(naType)
```

Comment:
https://golang.org/pkg/reflect/#MakeMapWithSize will be better

Reply:
Thanks. Will do